tests/main/snap-run-hook/task.yaml

```diff
@@ -0,0 +1,69 @@
+summary: Check that `snap run` can actually run hooks
+
+prepare: |
+    echo "Ensure we have a core snap with snap run"
+    $SPREAD_PATH/tests/lib/reset.sh
+    snap install --channel=candidate ubuntu-core
+    snapbuild $TESTSLIB/snaps/basic-hooks .
+    snap install basic-hooks_1.0_all.snap
+
+    # Use the in-tree snap-exec, not the one in the core snap. We'll do that via
+    # overlayfs and a copy.
+    tmp=$(mktemp -d)
+    overlay="$tmp/overlay"
+    work="$tmp/work"
+    mkdir "$overlay" "$work"
+    to_overlay="$(realpath /snap/ubuntu-core/current)"
+    mount -t overlayfs -o lowerdir=$to_overlay,upperdir=$overlay,workdir=$work overlayfs $to_overlay
+    cp /usr/lib/snapd/snap-exec $to_overlay/usr/lib/snapd/
+
+restore: |
+    rm basic-hooks_1.0_all.snap
+    umount "$(realpath /snap/ubuntu-core/current)"
+
+execute: |
+    # snap-confine can't run from /gopath
+    cd /home/test
+
+    # Note that `snap run` doesn't exit non-zero if the hook is missing, so we
+    # check stdout instead.
+
+    echo "Test that snap run can call valid hooks"
+
+    output="$(snap run --hook=apply-config basic-hooks)"
```

Comment:
Maybe a matter of personal preference but I find it slightly easier to merge this line and the following line into `if ! output="$(snap run --hook=apply-config basic-hooks)"; then echo "Failed to ...`  or even `output="$(snap run --hook=...) || (echo "Failed to ...")`